app/controller/command/commands/export.py

```diff
@@ -0,0 +1,192 @@
+                                  for ids in list(teams[0].members)]
+                        users = self.facade.query_or(User, params)
+                        return self.export_emails_helper(users)
+                    else:
+                        return self.permission_error, 200
+                else:  # if team name is not provided, export all emails
+                    if user_command.permissions_level == Permissions.admin:
+                        users = self.facade.query(User)
+                        return self.export_emails_helper(users)
+                    else:
+                        return self.permission_error, 200
+            except LookupError:
+                return self.lookup_error, 200
+        else:
+            return self.get_help(), 200
+
+    def export_emails_helper(self,
+                             users: list) -> ResponseTuple:
+        """
+        1. if team name not provided -> export emails of all users
+         as a string + names of the users who do not have an email
+
+        2. if team name provided -> export emails of all members of
+         the team + names of the users who do not have an email
+
+        """
+
+        if len(users) == 0:
+            return self.no_user_msg, 200
+
+        emails = []
+        ids_missing_emails = []
+
+        for i in range(len(users)):
+            if users[i].email == '':
+                ids_missing_emails.append(users[i].slack_id)
+                continue
+
+            emails.append(users[i].email)
+
+        # Check char count. Slack currently allows to send 16000 characters max
+        emails_str = ",".join(emails)
+
+        # Approximate char count for the names (20 chars)
+        # of the people who are missing emails
+        missing_emails_char_count = len(ids_missing_emails) * 20
+
+        char_limit_exceeded = False
+
+        if len(emails_str) + missing_emails_char_count > 15900:
+            emails_str = emails_str[:15900]
+            last_comma_idx = emails_str.rfind(',')
+            emails_str = emails_str[:last_comma_idx]
+            char_limit_exceeded = True
+
+        if len(ids_missing_emails) != 0:
+            if char_limit_exceeded:
+                ret = "```" + emails_str + "```\n\n" \
+                      + self.char_limit_exceed_msg \
+                      + "\n\nMembers who don't have an email: {}".format(
+                        ",".join(map(lambda u: f"<@{u}>", ids_missing_emails)))
+            else:
+                ret = "```" + emails_str + "```" \
+                      + "\n\nMembers who don't have an email: {}".format(
+                        ",".join(map(lambda u: f"<@{u}>", ids_missing_emails)))
+        else:
+            if char_limit_exceeded:
+                ret = "```" + emails_str + "```\n\n" \
+                      + self.char_limit_exceed_msg
+            else:
+                ret = "```" + emails_str + "```" \
+                      + "\n\n" + self.no_emails_missing_msg
```

Comment:
Interesting approach. Probably want to put the whole handling of char limits into another function.